src/base/memory_monitor_cl.cpp

```diff
@@ -0,0 +1,162 @@
+{
+  fprintf (stdout, "====================Cubrid Memmon====================\n");
+  fprintf (stdout, "Server Name: %s\n", server_info.name);
+  fprintf (stdout, "Total Memory Usage(KB): %lu\n\n", MMON_CONVERT_TO_KB_SIZE (server_info.total_mem_usage));
+  fprintf (stdout, "-----------------------------------------------------\n");
+}
+
+/*
+ * mmon_print_module_info -
+ *
+ * return:
+ *
+ *   module_info(in):
+ */
+void
+mmon_print_module_info (std::vector<MMON_MODULE_INFO> &module_info)
+{
+  const auto init_size_str = std::string {"Init Size(KB)"};
+  const auto cur_size_str = std::string {"Cur Size(KB)"};
+  const auto peak_size_str = std::string {"Peak Size(KB)"};
+  const auto resize_expand_str = std::string {"Resize Expand"};
+
+  for (const auto &m_info : module_info)
+    {
+      fprintf (stdout, "Module Name: %s\n\n", m_info.name);
+      fprintf (stdout, "%-13s\t: %17lu\n", init_size_str.c_str (), MMON_CONVERT_TO_KB_SIZE (m_info.stat.init_stat));
+      fprintf (stdout, "%-13s\t: %17lu\n", cur_size_str.c_str (), MMON_CONVERT_TO_KB_SIZE (m_info.stat.cur_stat));
+      fprintf (stdout, "%-13s\t: %17lu\n", peak_size_str.c_str (), MMON_CONVERT_TO_KB_SIZE (m_info.stat.peak_stat));
+      fprintf (stdout, "%-13s\t: %17u\n\n", resize_expand_str.c_str (), m_info.stat.expand_resize_count);
+
+      for (const auto &comp_info : m_info.comp_info)
+	{
+	  fprintf (stdout, "%s\n", comp_info.name);
+	  fprintf (stdout, "\t%-17s\t: %17lu\n", init_size_str.c_str (), MMON_CONVERT_TO_KB_SIZE (comp_info.stat.init_stat));
+	  fprintf (stdout, "\t%-17s\t: %17lu\n", cur_size_str.c_str (), MMON_CONVERT_TO_KB_SIZE (comp_info.stat.cur_stat));
+
+	  for (const auto &subcomp_info : comp_info.subcomp_info)
+	    {
+	      auto out_name = std::string {subcomp_info.name};
+	      out_name += "(KB)";
+	      fprintf (stdout, "\t  %-15s\t: %17lu\n", out_name.c_str (), MMON_CONVERT_TO_KB_SIZE (subcomp_info.cur_stat));
+	    }
+	  fprintf (stdout, "\t%-17s\t: %17lu\n", peak_size_str.c_str (), MMON_CONVERT_TO_KB_SIZE (comp_info.stat.peak_stat));
+	  fprintf (stdout, "\t%-17s\t: %17u\n\n", resize_expand_str.c_str (), comp_info.stat.expand_resize_count);
+	}
+      fprintf (stdout, "\n-----------------------------------------------------\n\n");
+    }
+}
+
+/*
+ * mmon_print_module_info_summary -
+ *
+ * return:
+ *
+ *   module_info(in):
+ */
+void
+mmon_print_module_info_summary (uint64_t server_mem_usage, std::vector<MMON_MODULE_INFO> &module_info)
+{
+  double cur_stat_ratio = 0.0;
+
+  fprintf (stdout, "Per Module Memory Usage (KB)\n");
+  fprintf (stdout, "\tModule\t\tModule\t\t\tModule\n");
+  fprintf (stdout, "\tIndex\t\tName\t\t\tUsage(%%)\n");
+
+  for (const auto &m_info : module_info)
+    {
+      if (server_mem_usage != 0)
+	{
+	  cur_stat_ratio = m_info.stat.cur_stat / (double) server_mem_usage;
+	}
+
+      fprintf (stdout, "\t%5d\t%-20s\t: %17lu(%3d%%)\n", mmon_convert_module_name_to_index (m_info.name),
```

Comment:
I think error handling for 'mmon_convert_module_name_to_index ()' is required, when it returns -1

Reply:
I added assert, and I think that's enough.